view/frontend/web/js/googlepay/button.js

```diff
@@ -273,6 +284,10 @@ define([
 
             onPaymentDataChanged: function (data) {
                 return new Promise((resolve, reject) => {
+                    if (virtualQuoteModel().getIsVirtual()) {
+                        resolve();
```

Comment:
There are a few issues with this early `resolve();` line which is causing the Google Pay UI dialog to fail from opening.

**Missing PaymentDataRequestUpdate payload**

As per Google Pay [documentation](https://developers.google.com/pay/api/web/reference/client#onPaymentDataChanged) the `resolve()` requires a payload passing through which is an object in the form of [PaymentDataRequestUpdate](https://developers.google.com/pay/api/web/reference/request-objects#PaymentDataRequestUpdate).

The properties of the object are all optional, therefor because this is a virtual quote we can pass in an empty object here. This might alter depending on your needs, for example if you support Google Pay promo codes you may want to pass through the `newTransactionInfo` property as a minimum.

**Early return needs adding in**

It's missing an early `return;` statement after the resolve line. Without it all the code below tries to run. Meaning it attempts to retrieve the `shippingAddress.countryCode` which doesn't exist on the `data` object, this in turn causes a fatal error as we can't access properties from an object that doesn't exist.

**Code changes:**

After implementing these changes it would look like the below:

```diff
- resolve();
+ resolve({});
+ return;
```